### PR TITLE
Use full path in mtl loading

### PR DESCRIPTION
**Pull request by vitvakatu**

### CODE REVIEW

File: src/obj.rs [hunk 1]
```diff
@@ -124,12 +125,15 @@ where
...
+            path: PathBuf::new(),
         }
     }
 
     pub fn load(path: &Path) -> io::Result<Obj<P>> {
         let f = File::open(path)?;
-        let obj = Obj::load_buf(&mut BufReader::new(f))?;
+        let mut obj = Obj::load_buf(&mut BufReader::new(f))?;
+        // unwrap is safe as we've read this file before
+        obj.path = path.parent().unwrap().to_owned();
```
Comment (kvark) [inline src/obj.rs:136]: why does it need `parent()` here?

File: src/obj.rs [hunk 2]
```diff
@@ -146,7 +150,9 @@ where
         let mut materials = HashMap::new();
 
         for m in &self.material_libs {
-            let file = match File::open(&m) {
+            let mut full_path = self.path.clone();
+            full_path.push(&m);
+            let file = match File::open(&full_path) {
```
Comment (kvark) [inline src/obj.rs]: could just do `self.path.join(&m)` if the path is for the folder

**Review by kvark — CHANGES_REQUESTED** [review]

File: src/obj.rs (hunk 1 above)
Comment (vitvakatu) [reply src/obj.rs:136]: Because the path is smth like "/long/path/file.obj", but we need only "/long/path/" part.

File: src/obj.rs (hunk 2 above)
Comment (vitvakatu) [reply src/obj.rs]: :+1:

**Review by kvark — APPROVED** [review]
Thanks!
bors r+

**Pull request merged**